### PR TITLE
feat (client): drop the DAL and the generator

### PR DESCRIPTION
**Pull request by kevin-dp**
This PR drops the DAL and the generator from the client.
DAL unit tests have been removed.
The remaining unit tests and e2e tests have been modified accordingly by replacing DAL queries by raw queries.

TODO:

- Update website (docs)

### CODE REVIEW

File: clients/typescript/src/util/types.ts [hunk 1]
```diff
@@ -19,7 +19,14 @@ export type Query = string
...
 export type RowId = number
-export type SqlValue = string | number | null | Uint8Array | bigint
+export type SqlValue =
+  | boolean
+  | string
+  | number
+  | Uint8Array
+  | undefined
+  | null
+  | bigint
```
Comment (msfstef) [inline clients/typescript/src/util/types.ts:29]: super nitty nit: perhaps having `bigint` under `number` looks better?

**Review by msfstef — APPROVED** [review]
Looks good! Nice to see so many things being removed!

Should this PR also remove Zod as a dependency? Or are we still using it for schema validation with the minimal DB schema provided? I'm wondering if we are using it minimally if it might be worth doing a custom implementation of a thin schema validator for rows just for the sake of removing the problematic dependency.

**Comment by kevin-dp** [comment]
> Looks good! Nice to see so many things being removed!
> 
> Should this PR also remove Zod as a dependency? Or are we still using it for schema validation with the minimal DB schema provided? I'm wondering if we are using it minimally if it might be worth doing a custom implementation of a thin schema validator for rows just for the sake of removing the problematic dependency.

We could make a follow-up PR that removes the Zod dependency altogether and replaces the usages of Zod by some custom validation code.

**Comment by KyleAMathews** [comment]
👋 we've been working the last month on a rebuild of the Electric server over at a temporary repo https://github.com/electric-sql/electric-next/

You can read more about why we made the decision at https://next.electric-sql.com/about

We're really excited about all the new possibilities the new server brings and we hope you'll check it out soon and give us your feedback.

We're now moving the temporary repo back here. As part of that migration we're closing all the old issues and PRs. We really appreciate you taking the time to investigate and create this improvement!

**Pull request closed without merging**